exchanges/huobi/huobi_wrapper.go

```diff
@@ -1962,8 +1962,13 @@ func (h *HUOBI) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pa
 		return nil, common.ErrFunctionNotSupported
 	case asset.Futures:
 		for i := range req.RangeHolder.Ranges {
+			// handling Huobi timezone retrieval differences
+			startDate := req.RangeHolder.Ranges[i].Start.Time.Add(-time.Hour * 16)
+			endDate := req.RangeHolder.Ranges[i].End.Time.Add(-time.Hour * 16)
```

Comment:
This is return empty candles
```
{Time:2023-03-24 00:00:00 +0000 UTC Open:0 High:0 Low:0 Close:0 Volume:0 ValidationIssues:} {Time:2023-03-25 00:00:00 +0000 UTC Open:0 High:0 Low:0 Close:0 Volume:0 ValidationIssues:} {Time:2023-03-26 00:00:00 +0000 UTC Open:0 High:0 Low:0 Close:0 Volume:0 ValidationIssues:} {Time:2023-03-27 00:00:00 +0000 UTC Open:0 High:0 Low:0 Close:0 Volume:0 ValidationIssues:} {Time:2023-03-28 00:00:00 +0000 UTC Open:0 High:0 Low:0 Close:0 Volume:0 ValidationIssues:} {Time:2023-03-29 00:00:00 +0000 UTC Open:0 High:0 Low:0 Close:0 Volume:0 ValidationIssues:} {Time:2023-03-30 00:00:00 +0000 UTC Open:0 High:0 Low:0 Close:0 Volume:0 ValidationIssues:}
```

and below

Reply:
Just an update on this. This occurs as a result of Huobi aligning candles to Singaporean time, while the new `request.CreateKlineRequest` truncates to a UTC time, misaligning the kline interval times, so no data can be populated, even though candles are returned.
Specifically, if I'm trying to get the candle at ticks `1680192000`, the request.CreateKlineRequest` will change it to `1680134400`. This then goes to `addPadding` which cannot match the times, so no kline data can be set, leaving all values as zero. 
Just trying to work around these restrictions

Reply:
😭